File: 02_Object_Oriented_Programming/05_Polymorphism_and_It's_types/04_Disadvantages/02_Performance_Overhead/javascript/main.js
const { Car } = require('../vehicle');

const startTime = process.hrtime.bigint();

// Dynamic method dispatch
const myVehicle = new Car();
myVehicle.start(); // Runtime resolves method implementation dynamically

const endTime = process.hrtime.bigint();
console.log(`Time taken for method dispatch: ${endTime - startTime} nanoseconds`);

/*
Output:
Starting a car
Time taken for method dispatch: XXXXX nanoseconds
*/

File: 02_Object_Oriented_Programming/05_Polymorphism_and_It's_types/04_Disadvantages/02_Performance_Overhead/python/main.py
import sys
import time
sys.path.append('..')
from vehicle import Car

start_time = time.time_ns()

# Dynamic method dispatch
my_vehicle = Car()
my_vehicle.start()  # Runtime resolves method implementation dynamically

end_time = time.time_ns()
print(f"Time taken for method dispatch: {end_time - start_time} nanoseconds")

"""
Output:
Starting a car
Time taken for method dispatch: XXXXX nanoseconds
"""

